airflow/providers/edge/plugins/edge_executor_plugin.py

```diff
@@ -0,0 +1,134 @@
+from typing import TYPE_CHECKING
+
+from connexion import FlaskApi
+from flask import Blueprint
+from flask_appbuilder import BaseView, expose
+from sqlalchemy import select
+
+from airflow.auth.managers.models.resource_details import AccessView
+from airflow.configuration import conf
+from airflow.exceptions import AirflowConfigException
+from airflow.models.taskinstance import TaskInstanceState
+from airflow.plugins_manager import AirflowPlugin
+from airflow.utils.session import NEW_SESSION, provide_session
+from airflow.utils.yaml import safe_load
+from airflow.www import utils as wwwutils
+from airflow.www.auth import has_access_view
+from airflow.www.constants import SWAGGER_BUNDLE, SWAGGER_ENABLED
+from airflow.www.extensions.init_views import _CustomErrorRequestBodyValidator, _LazyResolver
+
+if TYPE_CHECKING:
+    from sqlalchemy.orm import Session
+
+
+def _get_api_endpoints() -> Blueprint:
+    folder = Path(__file__).parents[1].resolve()  # this is airflow/providers/edge/
+    with folder.joinpath("openapi", "edge_worker_api_v1.yaml").open() as f:
+        specification = safe_load(f)
+    bp = FlaskApi(
+        specification=specification,
+        resolver=_LazyResolver(),
+        base_path="/edge_worker/v1",
+        options={"swagger_ui": SWAGGER_ENABLED, "swagger_path": SWAGGER_BUNDLE.__fspath__()},
+        strict_validation=True,
+        validate_responses=True,
+        validator_map={"body": _CustomErrorRequestBodyValidator},
+    ).blueprint
+    # Need to excemp CSRF to make API usable
+    from airflow.www.app import csrf
+
+    csrf.exempt(bp)
+    return bp
+
+
+# registers airflow/providers/edge/plugins/templates as a Jinja template folder
+template_bp = Blueprint(
+    "template_blueprint",
+    __name__,
+    template_folder="templates",
+)
+
+
+class EdgeWorkerJobs(BaseView):
+    """Simple view to show Edge Worker jobs."""
+
+    default_view = "jobs"
+
+    @expose("/jobs")
+    @has_access_view(AccessView.JOBS)
+    @provide_session
+    def jobs(self, session: Session = NEW_SESSION):
+        from airflow.providers.edge.models.edge_job import EdgeJobModel
+
+        jobs = session.scalars(select(EdgeJobModel).order_by(EdgeJobModel.queued_dttm)).all()
+        html_states = {
+            str(state): wwwutils.state_token(str(state)) for state in TaskInstanceState.__members__.values()
+        }
+        return self.render_template("edge_worker_jobs.html", jobs=jobs, html_states=html_states)
+
+
+class EdgeWorkerHosts(BaseView):
+    """Simple view to show Edge Worker status."""
+
+    default_view = "status"
+
+    @expose("/status")
+    @has_access_view(AccessView.JOBS)
+    @provide_session
+    def status(self, session: Session = NEW_SESSION):
+        from airflow.providers.edge.models.edge_worker import EdgeWorkerModel
+
+        hosts = session.scalars(select(EdgeWorkerModel).order_by(EdgeWorkerModel.worker_name)).all()
+        five_min_ago = datetime.now() - timedelta(minutes=5)
```

Comment:
Maybe we could make the 5 minutes definition configurable?

Reply:
This is at the moment just a small visual highlight if no heartbeat. There is no extended logic behind this. The other/usual job/task heartbeat that raises zombie tasks is independent of this. At the moment I would nto over-engineer these views with too many parameters. Otherwise I'd rather just remove the highlight :-D fr simplicity